src/main/java/org/togetherjava/formatter/tokenizer/Lexer.java

```diff
@@ -0,0 +1,48 @@
+package org.togetherjava.formatter.tokenizer;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.stream.Collectors;
+
+public class Lexer {
+    private static final Lexer instance = new Lexer();
```

Comment:
I dont see a reason yet to make this a Singleton, please reconsider your design choice here.